api-proposal.md

```diff
@@ -127,14 +194,16 @@ Return a list of all tags for autocomplete for search.
 
 ---
 
-### Authentication
+#### User
 
-Is the user authenticated?
+Returns information about the currently signed-in user.
 
-- GET `/api/v1/auth`
+- GET `/api/v1/user`
   ```
   {
+    isAuthenticated: bool //
     username: string // github username
```

Comment:
Maybe we can add email, avatar_url here(can't retrieve the information of every github user).

Reply:
Yeah, from your /user endpoint right now this is the data I see.

```javascript
{
  name: "sorahn",
  avatar_url: "https://avatars2.githubusercontent.com/u/588852?v=4",
  url: "https://api.github.com/users/sorahn",
  email: "daryl.roberts@gmail.com"
}
```

Reply:
Where will we keep this information? Are we adding users to the DB, or store them in memory?

Reply:
I'm not sure what the `email` field will be used for.
I agree that showing the `avatar_url` next to the username would be nice.

Reply:
I don't think we need to keep this information anywhere, IIUC it comes from github every time.

Reply:
Ah, the UI will talk directly to GH? 
I thought the UI is only talking to `/api/v1/user` to get the username + avatar url.

Reply:
I thought the server talks to github. Maybe I'm misunderstanding how the oauth stuff works.

Reply:
backend retrieve the infomation from github, and do some filter, then return to the frontend.

Reply:
@klion26 so it doesn't save anything right. just asks github every time?